pyproject.toml

```diff
@@ -1,4 +1,125 @@
+[tool.poetry]
+name = "dowhy"
+
+#
+# 0.0.0 is standard placeholder for poetry-dynamic-versioning
+# any changes to this should not be checked in
+#
+version = "0.0.0"
+description = "DoWhy is a Python library for causal inference that supports explicit modeling and testing of causal assumptions"
+authors = [
+    "PyWhy Community <amshar@microsoft.com>"
+]
+maintainers = []
+license = "MIT"
+documentation = "https://py-why.github.io/dowhy"
+repository = "https://github.com/pywhy/dowhy"
+classifiers = [  
+    'Development Status :: 4 - Beta',
+    'License :: OSI Approved :: MIT License',
+    'Programming Language :: Python :: 3.8',
+    'Programming Language :: Python :: 3.9',
+]
+keywords = [
+    'causality',
+    'machine-learning',
+    'causal-inference',
+    'statistics',
+    'graphical-model'
+]
+include = ['docs', 'tests', 'CONTRIBUTING.md']
+readme = 'README.rst'
+
+[tool.poetry-dynamic-versioning]
+enable = true
+vcs = "git"
+
+[tool.poetry-dynamic-versioning.substitution]
+files = ["dowhy/__init__.py"]
+
+#
+# Dependency compatibility notes:
+# * xgboost requires Python >=3.7,<3.11
+# * pygraphviz requires Python >=3.8
+# * networkx requires Python >= 3.8
+# * llvmlite requires Python >=3.6,<3.11
+#
+[tool.poetry.dependencies]
+python = ">=3.8,<3.10"
+scipy = "^1.8.1"
+statsmodels = "^0.13.2"
+numpy = "^1.23.1"
+pandas = "^1.4.3"
+networkx = "^2.8.5"
+sympy = "^1.10.1"
+scikit-learn = "1.0.2"
+pydot = "^1.4.2"
+joblib = "^1.1.0"
+# causalml is wired to use llvmlite 0.36
+llvmlite = "^0.36.0"
```

Comment:
should we put llmvlite as a dev-dependency? It is not used by dowhy, but only relevant if someone is using causalml.
Another option is to mark it as an optional dependency.

Reply:
I can do that; if it's optional we would probably add it as a devDep and mark it as optional in the main deps. Should causalml be optional as well?

Reply:
yes, causalml should be optional too.